packages/block-editor/src/components/block-list/style.scss

```diff
@@ -118,6 +130,19 @@
 .block-editor-block-list__layout .block-editor-block-list__block {
 	position: relative;
 
+	// Re-enable text-selection on editable blocks.
+	user-select: text;
+
+	// Hide the select style pseudo element as it interferes with the style.
+	&.is-partially-selected::after {
+		height: 0;
```

Comment:
This is a good change which I think fixes the issues reported by @ntsekouras. Specifically instead of removing the pseudo element, it's made invisible. In practice for me that means I can hover the drag handle to highlight the blocks again, and click between them to de-select.